services/report-charts/src/weaver_report/compare.py

```
@@ -703,3 +703,247 @@ def _memory_edge_weight_tabular(
+# ---------------------------------------------------------------------------
+# Nap + sleep pair charts (read trace spans directly via the same
+# helpers the single-run nap charts use).
+# ---------------------------------------------------------------------------
+
+
+def _nap_records_safe(run: Run, side: str) -> list[dict[str, Any]]:
+    """Same as charts.nap._nap_records but raises with a side-tagged
+    ChartUnavailable instead of the bare one, so paired-chart errors
+    name which side lacked data."""
+    if run.trace_path is None:
+        raise ChartUnavailable(f"side {side}: no trace file")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)